python/lsst/cell_coadds/_fits.py

```diff
@@ -374,11 +432,59 @@ def writeMultipleCellCoaddAsFits(
     metadata : `~lsst.daf.base.PropertySet`, optional
         Additional metadata to write to the FITS file.
 
+    Returns
+    -------
+    hdu_list : `~astropy.io.fits.HDUList`
+        The FITS file as an HDUList.
+
     Notes
     -----
     Changes to this function that modify the way the file is written to disk
     must be accompanied with a change to FILE_FORMAT_VERSION.
     """
+    # Create metadata tables:
+    # 1. Visit table containing information about the visits.
+    # 2. Cell table containing info about the visit+detector for each cell.
+    visit_records: list[Any] = []
+    cell_records: list[Any] = []
+    instrument_set = set()
+    for cell_id, single_cell_coadd in multiple_cell_coadd.cells.items():
+        for observation_id in single_cell_coadd.inputs:
+            visit_records.append(
+                (observation_id.visit, observation_id.physical_filter, observation_id.day_obs)
+            )
+            cell_records.append(
+                (cell_id.x, cell_id.y, observation_id.visit, observation_id.detector, observation_id.packed)
+            )
+            instrument_set.add(observation_id.instrument)
+
+    assert len(instrument_set) == 1, "All cells must have the same instrument."
+    instrument = instrument_set.pop()
+
+    visit_recarray = np.rec.fromrecords(
+        recList=sorted(set(visit_records), key=lambda x: x[0]),  # Sort by visit.
+        formats=None,  # formats has specified to please mypy. See numpy#26376.
+        names=(
+            "visit",
+            "physical_filter",
+            "day_obs",
+        ),
+    )
+    cell_recarray = np.rec.fromrecords(
+        recList=cell_records,
+        formats=None,  # formats has specified to please mypy. See numpy#26376.
```

Comment:
Someday when my [fix upstream](https://github.com/numpy/numpy/pull/26377) makes its way to our `rubin-env` we can remove it specifying `formats=None`. Today is not that day.